### PR TITLE
fronend code updates first iteration

### PR DESCRIPTION
**Pull request by achen2401**
To try this out:

first run `npm install` to install all dependencies
next run `npm run build` to get the initial build of frontend code

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,6 @@
 *__pycache__
 client_secrets.json
 dashboard/static/js
+dashboard/templates/index.html
```
Comment (pbugni) [inline .gitignore:5]: @achen2401 are you sure about ignoring this `index.html` file?

Comment (achen2401) [reply .gitignore:5]: yes, the frontend code should auto-generate the file with required js files injected.  I made use of this webpack plugin `HtmlWebpackPlugin`, specified in webpack.config.js

File: dashboard/static/src/index.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title><%= htmlWebpackPlugin.options.title %></title>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.3.0/css/font-awesome.min.css">
```
Comment (ivan-c) [inline dashboard/static/src/index.html:8]: Is there a reason why we would or wouldn't want to use `package.json` exclusively (for all libs)?

File: dashboard/static/src/js/components/Table.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  componentDidMount() {
+    sendRequest("./Patient").then(response => {
+      let rawData = {};
+      if (response) {
+        rawData = JSON.parse(response);
+      }
+      let dataSet = [];
+      if (rawData["entry"]) {   
+        (rawData["entry"]).forEach(function(item) {
+            var nameNode = item.resource && item.resource.name ? item.resource.name[0] : null;
```
Comment (ivan-c) [inline dashboard/static/src/js/components/Table.js:24]: Was just reading about "destructuring assignment" looks complicated, but maybe worth it for even more deeply nested FHIR?

https://blog.opendigerati.com/how-to-extract-nested-data-from-arrays-and-objects-with-one-line-of-code-using-destructuring-77f7fc25230c?gi=57355b814e64
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

File: dashboard/static/src/index.html (hunk 2 above)
Comment (achen2401) [reply dashboard/static/src/index.html:8]: Hi, ivan, I actually thought about that.  I will try to do that when I have more time.  I also am not sure if bundle size will become a concern...I will do more research on bundle size management before falling back to using package.json for all libs.

File: dashboard/static/src/js/components/Table.js (hunk 3 above)
Comment (achen2401) [reply dashboard/static/src/js/components/Table.js:24]: that sounds great, ivan, will it still work for null object though, e.g. {a: name, b: phone} = obj where obj is null.  It seemed to break for me.

File: dashboard/static/src/index.html (hunk 2 above)
Comment (pbugni) [reply dashboard/static/src/index.html:8]: @achen2401 when you say you'll try when you have more time, I assume that means beyond this PR, as in we should merge and you'll address later?

Comment (achen2401) [reply dashboard/static/src/index.html:8]: yes, Paul, sorry I wasn't clear.  I am need to switch gear to work on Fenway project.  I will come back to address this afterwards.

File: dashboard/static/src/js/components/Table.js (hunk 3 above)
Comment (ivan-c) [reply dashboard/static/src/js/components/Table.js:24]: Ah darn, I would've thought it could handle missing/null attributes

**Pull request merged**